cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1115,27 +1120,68 @@ function Verify-WorkerServices {
   Log_Todo "run more verification commands."
 }
 
+function DownloadAndInstall-Crictl {
+  $CRICTL_VERSION = "v1.16.1"
+  $CRICTL_SHA256 = "7d092dcb3b1af2edf75477d5d049a70e8c0d1ac8242b1dff2de7e6aa084e3615"
+
+  # Assume that presence of crictl.exe indicates that the crictl binaries
+  # were already previously downloaded to this node.
+  if (-not (ShouldWrite-File ${env:NODE_DIR}\crictl.exe)) {
+    return
+  }
+  $tmp_dir = 'C:\crictl_tmp'
+  New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
+
+  $url = ('https://github.com/kubernetes-sigs/cri-tools/releases/' +
+      'download/' + $CRICTL_VERSION + '/crictl-' +
+      $CRICTL_VERSION + '-windows-amd64.tar.gz')
+  MustDownload-File `
+      -URLs $url `
+      -OutFile $tmp_dir\crictl.tgz `
+      -Hash $CRICTL_SHA256 `
+      -Algorithm SHA256
+
+  Push-Location $tmp_dir
+  # tar can only extract in the current directory.
+  tar -xvf $tmp_dir\crictl.tgz
+  Move-Item -Force crictl.exe ${env:NODE_DIR}\
+  if (${env:CONTAINER_RUNTIME_ENDPOINT}) {
+    crictl.exe config runtime-endpoint ${env:CONTAINER_RUNTIME_ENDPOINT}
+  }
+  Pop-Location
+  Remove-Item -Force -Recurse $tmp_dir
+}
+
 # Pulls the infra/pause container image onto the node so that it will be
 # immediately available when the kubelet tries to run pods.
 # TODO(pjh): downloading the container container image may take a few minutes;
 # figure out how to run this in the background while perform the rest of the
 # node startup steps!
 function Pull-InfraContainer {
+  $COMMAND = "docker"
+  $LIST = "images"
+  $PULL = "pull"
+  $INSPECT = "inspect"
+  if (${env:CONTAINER_RUNTIME} -ne "docker") {
+    $COMMAND = "crictl"
```

Comment:
Crictl talks with dockershim instead of dockerd. So once kubelet is started, crictl will work.

However, here kubelet hasn't started yet, we have to use docker.

Reply:
Ah, I see. Can you add a comment explaining that?

Reply:
Actually we moved PullInfraContainer to the end after kubelet started, so I think we should be able to use crictl now.

I'll still add comment to emphasize that the order needs to be kept in the future.